FlagsmithClient/Classes/Internal/SSEManager.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  SSEManager.swift
+//  FlagsmithClient
+//
+//  Created by Gareth Reese on 13/09/2024.
+//
+
+import Foundation
+#if canImport(FoundationNetworking)
+    import FoundationNetworking
+#endif
+
+/// SSEManager handles interaction with the Flagsmith SSE real-time API.
+/// It manages the connection to the SSE endpoint, processes incoming events,
+/// and handles reconnection logic with a backoff strategy.
+final class SSEManager: NSObject, URLSessionDataDelegate, @unchecked Sendable {
+    private var _session: URLSession!
+    private var session: URLSession {
+        get {
+            propertiesSerialAccessQueue.sync { _session }
+        }
+        set {
+            propertiesSerialAccessQueue.sync(flags: .barrier) {
+                _session = newValue
+            }
+        }
+    }
+
+    private var _dataTask: URLSessionDataTask?
+    private var dataTask: URLSessionDataTask? {
+        get {
+            propertiesSerialAccessQueue.sync { _dataTask }
+        }
+        set {
+            propertiesSerialAccessQueue.sync(flags: .barrier) {
+                _dataTask = newValue
+            }
+        }
+    }
+
+    /// Base `URL` used for requests.
+    private var _baseURL = URL(string: "https://realtime.flagsmith.com/")!
+    var baseURL: URL {
+        get {
+            propertiesSerialAccessQueue.sync { _baseURL }
+        }
+        set {
+            propertiesSerialAccessQueue.sync {
+                _baseURL = newValue
+            }
+        }
+    }
+
+    /// API Key unique to an organization.
+    private var _apiKey: String?
+    var apiKey: String? {
+        get {
+            propertiesSerialAccessQueue.sync { _apiKey }
+        }
+        set {
+            propertiesSerialAccessQueue.sync {
+                _apiKey = newValue
+            }
+        }
+    }
+
+    var isStarted: Bool {
+        return completionHandler != nil
+    }
+
+    private var completionHandler: CompletionHandler<FlagEvent>?
+    private let serialAccessQueue = DispatchQueue(label: "sseFlagsmithSerialAccessQueue", qos: .default)
+    let propertiesSerialAccessQueue = DispatchQueue(label: "ssePropertiesSerialAccessQueue", qos: .default)
+    private let reconnectionDelay = ReconnectionDelay()
+
+    override init() {
+        super.init()
+        let configuration = URLSessionConfiguration.default
+        session = URLSession(configuration: configuration, delegate: self, delegateQueue: OperationQueue.main)
+    }
+
+    // Helper function to process SSE data
+    internal func processSSEData(_ data: String) {
+        // Split the data into lines and decode the 'data:' lines from JSON into FlagEvent objects
+        let lines = data.components(separatedBy: "\n")
+        for line in lines where line.hasPrefix("data:") {
+            let json = line.replacingOccurrences(of: "data:", with: "")
```

Comment:
This is mostly just my pythonic brain thinking that defining a variable called `json` is a no-no, but regardless, I think we can be more explicit here

```suggestion
            let jsonMessageData = line.replacingOccurrences(of: "data:", with: "")
```